### ISSUE TITLE
Magisk Manager update V6 to V7 not so easy

### ISSUE THREAD

**Issue by wescande** [topjohnwu/Magisk#1048]
**TL:DR;**
Hiding Magisk Manager (by repacking the package) prevent Manager && magisk update

I recently had a popup saying a new version is release for both Manager and Magisk. The popup ask to install, so I did. It install the new Magisk Manager V7 along to old one ! But app menu was missing in the V7 (superuser/magiskHide/module/donwloads/logs ). I tried to update Magisk from 18 to 18.1 and there was no option for direct Install neither for Install to inactive Slot. I try to reboot and then the magisk manager V7 was uninstall. Launching the old one was always saying to install V7.

Finally I just use the Restore Magisk Manager option in settings menu and everything was able to work. I updated to V7 immediaty and to 18.1 in the same way.

I think it's a similar thing happened to #1039

So could it be possible to check it before installing update ? if yes we could imagine some scenario like
- Popup prevent update and saying to restore magisk (easy way)
- Popup warning about an automatic script that will restore then update then re-hide Magisk Manager (nice work-around way)
- or maybe it's should already work AND/OR it's supposed to not be like this (Voodoo level)

Thanks for all your works, it's amazing to use it !

**Comment by topjohnwu** [comment]
There were some bugs in old versions regarding upgrading repackaged Magisk Managers, it is fixed in v7.0.0, and there is no real good way to fix those on old versions other than manual removing and update
